### PR TITLE
[vcpkg baseline][pcre2] fix checksum and update to 10.39

### LINKED ISSUE #21255: pcre2[core]:x64-windows -> 10.37#1 build failure
Computing installation plan...
The following packages will be built and installed:
    pcre2[core]:x64-windows -> 10.37#1
Detecting compiler hash for triplet x64-windows...
Restored 0 packages from C:\Users\igor\AppData\Local\vcpkg\archives in 173.2 us. Use --debug to see more details.
Starting package 1/1: pcre2:x64-windows
Building package pcre2[core]:x64-windows...
-- Downloading https://github.com/PhilipHazel/pcre2/archive/pcre2-10.37.tar.gz -> PhilipHazel-pcre2-pcre2-10.37.tar.gz...
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'manifests' = off
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Downloading https://github.com/PhilipHazel/pcre2/archive/pcre2-10.37.tar.gz
[DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(656)
Error: Failed to download from mirror set:
File does not have the expected hash:
             url : [ https://github.com/PhilipHazel/pcre2/archive/pcre2-10.37.tar.gz ]
       File path : [ C:\Libraries\vcpkg\downloads\PhilipHazel-pcre2-pcre2-10.37.tar.gz.8880.part ]
   Expected hash : [ f91760a8e0747f52211612fb0e134d685e224d16bd884eb574718d077a586b1fd7b6435d4e3b75c879b12e02b252467ecc28cdc4bc2903c783dacab089f99c99 ]
     Actual hash : [ f15357d8f5400739f26a32fe03a0759607fd7a282a1a8a2bb169b971a8a00330eff3ba465842c4668ad968e3fd1fff4dc7f0631d0af8416947070ff7cd056e6f ]

[DEBUG] Exiting after %s (%d us)

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:84 (message):
      
      Failed to download file with error: 1
      If you use a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. You are using Fiddler. Currently a bug (https://github.com/microsoft/vcpkg/issues/17752)
         will set HTTPS_PROXY to `https://fiddler_address:port` which lead to problem 1 above.
         Workaround is open Windows 10 Settings App, and search for Proxy Configuration page,
         Change `http=address:port;https=address:port` to `address`, and fill the port number.
      
      3. You proxy's remote server is out of service.
      
      In future vcpkg releases, if you are using Windows, you no longer need to set
      HTTP(S)_PROXY environment variables. Vcpkg will simply apply Windows IE Proxy
      Settings set by your proxy software. See (https://github.com/microsoft/vcpkg-tool/pull/49)
      and (https://github.com/microsoft/vcpkg-tool/pull/77)
      
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:292 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:173 (vcpkg_download_distfile)
  ports/pcre2/portfile.cmake:3 (vcpkg_from_github)
  scripts/ports.cmake:142 (include)

Error: Building package pcre2:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  package: pcre2[core]:x64-windows -> 10.37#1
  vcpkg version: 2021-11-02-af04ebf6274fd6f7a941bff4662b3955c64f6f42
  vcpkg-tool version: 66c39e113 2021-11-05 (2 days ago)

Additionally, attach any relevant sections from the log files above.

### PR DESCRIPTION
**Pull request by camuward**
fix a [checksum mismatch]() for 10.37, and update to 10.39

- #### What does your PR fix?  
  fixes the invalid checksum from #21193

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

**Comment by JackBoosY** [comment]
I'm going to make a PR to fix the SHA512 issue, untill I saw this PR.

**Comment by dg0yt** [comment]
Port pcre2 installs `share/pcre2/pcre2-config.cmake`. What problem is to be solved by the cmake wrapper?

**Comment by JackBoosY** [comment]
> Port pcre2 installs `share/pcre2/pcre2-config.cmake`. What problem is to be solved by the cmake wrapper?

The pcre2-config.cmake provided by the upstream is not mature, so I choose to disable them.

File: ports/pcre2/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,16 @@
 {
   "name": "pcre2",
-  "version-string": "10.37",
-  "port-version": 1,
-  "description": "PCRE2 is a re-working of the original Perl Compatible Regular Expressions library",
-  "homepage": "https://pcre.org/"
+  "version-string": "10.39",
```
Comment (c72578) [inline ports/pcre2/vcpkg.json:3]:
```suggestion
  "version": "10.39",
```

**Comment by BillyONeal** [comment]
Thanks for the SHA fix!

**Pull request merged**